appmap/_implementation/detect_enabled.py

```diff
@@ -0,0 +1,135 @@
+
+    @classmethod
+    def should_enable(cls):
+        """
+        Should recording be enabled for the current recording method?
+        """
+        message, enabled = cls.detect_should_enable()
+        if enabled:
+            logger.warning(
+                dedent(f"AppMap recording is enabled because {message}")
+            )
+        return enabled
+
+    @classmethod  
+    def detect_should_enable(cls):
+        # explicitly disabled
+        if 'APPMAP' in os.environ and os.environ['APPMAP'] == 'false':
+            return ["APPMAP=false", False]
+
+        # recording method explicitly disabled
+        if cls._recording_method:
+            for recording_method in RECORDING_METHODS:
+                if recording_method == cls._recording_method.lower():
+                    env_var = '_'.join([ 'APPMAP', 'RECORD', cls._recording_method.upper() ])
+                    if env_var in os.environ:
+                        if os.environ[env_var] == 'false':
+                            return [f"{env_var}=false", False]
+                        elif os.environ[env_var] == 'true':
+                            return [f"{env_var}=true", True]
+
+        # recording method enabled
+        if cls._recording_method and cls._recording_method.lower() in RECORDING_METHODS:
+            return [f"{cls._recording_method} is enabled", True]
+
+        # it's flask
+        message, should_enable = cls.is_flask_and_should_enable()
+        if should_enable:
+            return [message, should_enable]
+
+        # it's django
+        message, should_enable = cls.is_django_and_should_enable()
+        if should_enable:
+            return [message, should_enable]
+
+        # no recording method set
+        if not cls._recording_method:
+            return ["no recording method is set", False]
+
+        return ["it is not enabled by any configuration or framework", False]
+
+    @classmethod
+    def is_flask_and_should_enable(cls):
+        if 'FLASK_DEBUG' in os.environ and os.environ['FLASK_DEBUG'] == '1':
+            return [f"FLASK_DEBUG={os.environ['FLASK_DEBUG']}", True]
+
+        if 'FLASK_ENV' in os.environ and os.environ['FLASK_ENV'] == 'development':
+            return [f"FLASK_ENV={os.environ['FLASK_ENV']}", True]
+
+        return ["it is not Flask", False]
+
+    @classmethod
+    def is_django_and_should_enable(cls):
+        if 'DJANGO_SETTINGS_MODULE' in os.environ and os.environ['DJANGO_SETTINGS_MODULE'] != '':
+            try:
+                settings = importlib.import_module(os.environ['DJANGO_SETTINGS_MODULE'])
+            except Exception as exn:
+                settings = None
+                return ["couldn't load DJANGO_SETTINGS_MODULE={os.environ['DJANGO_SETTINGS_MODULE']}", False]
+
+            if settings:
+                try:
+                    if settings.DEBUG == True:
+                        return [f"{os.environ['DJANGO_SETTINGS_MODULE']}.DEBUG={os.environ['DEBUG']}", True]
+                except:
```

Comment:
Better not to write bare `except`s. This code raises an exception when accessing `os.environ['DEBUG']`, and causes the function to incorrectly return `["it is not Django", False]`.